app/components/cityView.jsx

```diff
@@ -0,0 +1,172 @@
+"use client";
+import { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  deleteCity,
+  fetchForecast,
+  resetList,
+} from "@/store/slices/forecastSlice";
+
+import {
+  Sparklines,
+  SparklinesLine,
+  SparklinesReferenceLine,
+} from "react-sparklines";
+
+export default function CityView() {
+  const { forecastData, loading, error, cities } = useSelector(
+    (state) => state.forecast
+  );
+
+  const dispatch = useDispatch();
+
+  const [query, setQuery] = useState("");
+
+  useEffect(() => {}, []);
+
+  const handleSearch = () => {
+    if (!query || query.trim() === "") {
+      alert("Search cannot be empty");
+      setQuery("");
+    } else {
+      dispatch(fetchForecast(query));
+      setQuery("");
```

Comment:
DRY
create a function resetSearchBox that calls setQuery. Will help with readability also.